### PR TITLE
Converts no slip into a trait

### PR DESCRIPTION
**Pull request by BiancaWilkson**
## What Does This PR Do

Converts all the uses of the NOSLIP flag to instead use a trait. This is going to be very conflicted with #23000 so I'll mark it as draft until I get that merge conflict fixed.
## Why It's Good For The Game

Traits are helpful, and this allows coders/admins to easily give someone noslip without them needing to have boots with a specific flag.
## Testing

Lots of walking on, and slipping on soap with various noslips and magboots.
## Changelog
NPFC

### CODE REVIEW

File: code/modules/clothing/shoes/magboots.dm [hunk 1]
```diff
@@ -34,14 +36,18 @@
...
 /obj/item/clothing/shoes/magboots/proc/toggle_magpulse(mob/user, forced)
-	if(magpulse)
-		flags &= ~NOSLIP
+	if(magpulse) //magpulse and no_slip will always be the same value unless VV happens
+		START_PROCESSING(SSobj, src) //Gravboots
+		REMOVE_TRAIT(user, TRAIT_NOSLIP, UID())
 		slowdown = slowdown_passive
 	else
-		flags |= NOSLIP
+		STOP_PROCESSING(SSobj, src)
```
Comment (DGamerL) [inline code/modules/clothing/shoes/magboots.dm:44]: This doesn't process anymore

File: code/modules/clothing/spacesuits/alien_suits.dm [hunk 2]
```diff
@@ -207,12 +207,11 @@
 	species_restricted = list("Vox")
 	sprite_sheets = list(
 		"Vox" = 'icons/mob/clothing/species/vox/feet.dmi')
+	multiple_icons = FALSE
 
 /obj/item/clothing/shoes/magboots/vox/attack_self(mob/user)
 	if(magpulse)
-		flags &= ~NOSLIP
-		magpulse = FALSE
-		flags |= NODROP
```
Comment (DGamerL) [inline code/modules/clothing/spacesuits/alien_suits.dm:215]: ... did I break this?
Anyways, good fixing 👍

**Review by DGamerL — CHANGES_REQUESTED** [review]
Looks good, just that processing bit

File: code/modules/clothing/shoes/magboots.dm [hunk 3]
```diff
@@ -61,7 +65,7 @@
 		REMOVE_TRAIT(user, TRAIT_MAGPULSE, "magboots")
 
 /obj/item/clothing/shoes/magboots/negates_gravity()
-	return flags & NOSLIP
+	return magpulse
```
Comment (DGamerL) [inline code/modules/clothing/shoes/magboots.dm:68]: This also should still be a noslips check most likely

Comment (BiancaWilkson) [reply code/modules/clothing/shoes/magboots.dm:68]: For magboots, magpulse and no_slip are always going to be the same value, so there'd be no difference

Comment (Contrabang) [reply code/modules/clothing/shoes/magboots.dm:68]: Agreed, it should be consistent, even if they're supposed to always be the same var

File: code/modules/clothing/spacesuits/alien_suits.dm (hunk 2 above)
Comment (BiancaWilkson) [reply code/modules/clothing/spacesuits/alien_suits.dm:215]: Seems to be working fine

**Review by DGamerL — APPROVED** [review]

File: code/modules/clothing/clothing.dm [hunk 4]
```diff
@@ -480,6 +481,20 @@
...
+		return
+	ADD_TRAIT(user, TRAIT_NOSLIP, UID())
+
+/obj/item/clothing/shoes/dropped(mob/user)
+	..()
+	if(!no_slip)
+		return
+	var/mob/living/carbon/human/H = user
+	if(H.get_item_by_slot(SLOT_HUD_SHOES) == src)
+		REMOVE_TRAIT(H, TRAIT_NOSLIP, UID())
```
Comment (lewcc) [inline code/modules/clothing/clothing.dm:496]: Is there a possibility that they will have already been removed from this slot by the time this code is called?

File: code/modules/mod/modules/modules_antag.dm [hunk 5]
```diff
@@ -143,10 +143,10 @@
 	origin_tech = "syndicate=1"
 
 /obj/item/mod/module/noslip/on_suit_activation()
-	mod.boots.flags |= NOSLIP
+	ADD_TRAIT(mod.wearer, TRAIT_NOSLIP, src)
```
Comment (lewcc) [inline code/modules/mod/modules/modules_antag.dm:146]: any reason for using src over uid() here?

**Review by S34NW — CHANGES_REQUESTED** [review]
Please address review

File: code/modules/clothing/clothing.dm (hunk 4 above)
Comment (BiancaWilkson) [reply code/modules/clothing/clothing.dm:496]: I do not think so? I mostly copied the way judo belt/boxing gloves remove a trait, and dropping it seemed to work properly when I was testing it.

File: code/modules/mod/modules/modules_engineering.dm [hunk 6]
```diff
@@ -57,15 +57,15 @@
 	. = ..()
 	if(!.)
 		return
-	mod.boots.flags |= NOSLIP
+	ADD_TRAIT(mod.wearer, TRAIT_NOSLIP, src)
```
Comment (S34NW) [inline code/modules/mod/modules/modules_engineering.dm:60]:
```suggestion
	ADD_TRAIT(mod.wearer, TRAIT_NOSLIP, UID())
```

File: code/modules/mod/modules/modules_engineering.dm [hunk 7]
```diff
@@ -57,15 +57,15 @@
...
 	mod.slowdown += slowdown_active
 	ADD_TRAIT(mod.wearer, TRAIT_MAGPULSE, "magbooted")
 
 /obj/item/mod/module/magboot/on_deactivation(display_message = TRUE, deleting = FALSE)
 	. = ..()
 	if(!.)
 		return
-	mod.boots.flags ^= NOSLIP
+	REMOVE_TRAIT(mod.wearer, TRAIT_NOSLIP, src)
```
Comment (S34NW) [inline code/modules/mod/modules/modules_engineering.dm:68]:
```suggestion
	REMOVE_TRAIT(mod.wearer, TRAIT_NOSLIP, UID())
```

**Review by S34NW — APPROVED** [review]

**Pull request merged**